pkg/sql/parser/eval.go

```diff
@@ -2088,6 +2088,11 @@ func (expr *FuncExpr) Eval(ctx *EvalContext) (Datum, error) {
 
 	res, err := expr.fn.fn(ctx, args)
 	if err != nil {
+		// If we are facing a retry error, propagate it unchanged, so that
+		// the executor can see it with the right type.
+		if _, ok := err.(*roachpb.RetryableTxnError); ok {
```

Comment:
This special casing is problematic. Maybe you could instead use `errors.Wrapf`:
`errors.Wrapf(err, "%s()", expr.Func)`

And then where we check errors, you could use:

`errors.Cause(err)`